### PR TITLE
cxx-qt-gen: add prop/prop_mut/set_prop for Rust fields as well

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Note that the prop_mut is not unsafe as there is no signal that should be emitted.

Requires #267

### CODE REVIEW

File: examples/demo_threading/rust/src/lib.rs [hunk 1]
```diff
@@ -299,28 +299,18 @@ mod ffi {
...
-                                    // TODO: for now we use the unsafe rust_mut() API
-                                    // later there will be getters and setters for the properties
-                                    unsafe {
-                                        // Process the new data from the background thread
-                                        if let Some(data) = qobject_energy_usage
+                                    // Process the new data from the background thread
+                                    if let Some(data) =
+                                        qobject_energy_usage.as_mut().qt_rx_mut().try_iter().last()
+                                    {
+                                        // Here we have constructed a new Data struct so can consume it's values
```
Comment (Be-ing) [inline examples/demo_threading/rust/src/lib.rs:306]:
```suggestion
                                        // Here we have constructed a new Data struct so can consume its values
```

File: examples/demo_threading/rust/src/lib.rs [hunk 2]
```diff
@@ -299,28 +299,18 @@ mod ffi {
...
+                                        // for other uses we could have passed an Enum across the channel
```
Comment (Be-ing) [inline examples/demo_threading/rust/src/lib.rs:307]:
```suggestion
                                        // For other uses, we could have passed an Enum across the channel
```

File: examples/demo_threading/rust/src/lib.rs (hunk 1 above)
Comment (ahayzen-kdab) [reply examples/demo_threading/rust/src/lib.rs]: Done

File: examples/demo_threading/rust/src/lib.rs (hunk 2 above)
Comment (ahayzen-kdab) [reply examples/demo_threading/rust/src/lib.rs]: Done

File: examples/demo_threading/rust/src/lib.rs [hunk 3]
```diff
@@ -299,28 +299,18 @@ mod ffi {
...
-                                    // later there will be getters and setters for the properties
-                                    unsafe {
-                                        // Process the new data from the background thread
-                                        if let Some(data) = qobject_energy_usage
+                                    // Process the new data from the background thread
+                                    if let Some(data) =
+                                        qobject_energy_usage.as_mut().qt_rx_mut().try_iter().last()
+                                    {
+                                        // Here we have constructed a new Data struct so can consume its values.
+                                        // Foror other uses, we could have passed an Enum across the channel
```
Comment (Be-ing) [inline examples/demo_threading/rust/src/lib.rs:307]:
```suggestion
                                        // For other uses, we could have passed an Enum across the channel
```

still typos :)

**Review by Be-ing — APPROVED** [review]

**Pull request merged**